bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -180,6 +194,13 @@
 
         this.ledgerId = ledgerId;
 
+        if (clientCtx.getConf().enableStickyReads
+                && getLedgerMetadata().getEnsembleSize() == getLedgerMetadata().getWriteQuorumSize()) {
+            stickyBookieIndex = ThreadLocalRandom.current().nextInt() % getLedgerMetadata().getEnsembleSize();
```

Comment:
it might be better to add a method in `EnsemblePlacementPolicy`

```
int getStickyReadBookieIndex(LedgerMetadata metadata);
```

it can pick up a bookie randomly by default. for rack-aware and region-aware placement, it can be implemented using network locations.